ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh

```diff
@@ -254,6 +254,61 @@ EOF
     ansible-playbook -i "${SHARED_DIR}/ansible-hosts" /tmp/repo.yaml -vvv
 }
 
+# Do sdn migration to ovn since sdn is not supported from 4.17 version
+function sdn2ovn(){
+    TARGET=${TARGET:-OVNKubernetes}
+    timeout 600s bash <<EOT
+    until
+       oc wait co --all --for='condition=AVAILABLE=True' --timeout=10s && \
+       oc wait co --all --for='condition=PROGRESSING=False' --timeout=10s && \
+       oc wait co --all --for='condition=DEGRADED=False' --timeout=10s;
+    do
+       sleep 10
+       echo "Some ClusterOperators Degraded=False,Progressing=True,or Available=False";
+    done
+EOT
+    oc patch network.operator.openshift.io cluster --type='merge'  -p='{"spec":{"defaultNetwork":{"ovnKubernetesConfig":{"ipv4":{"internalJoinSubnet": "100.65.0.0/16"}}}}}' 
+    oc patch network.operator.openshift.io cluster --type='merge'  -p='{"spec":{"defaultNetwork":{"ovnKubernetesConfig":{"ipv4":{"internalTransitSwitchSubnet": "100.85.0.0/16"}}}}}' 
+    oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"network.openshift.io/network-type-migration":""}},"spec":{"networkType":"OVNKubernetes"}}'
+    timeout 300s bash <<EOT
+    until 
+       oc get network -o yaml | grep NetworkTypeMigrationInProgress > /dev/null
+    do
+       echo "Migration is not started yet"
+       sleep 10
+    done
+EOT
+    echo "Start Live Migration process now"
+
+    # Wait for the live migration to fully complete
+
+    timeout 3600s bash <<EOT
+    until 
+       oc get network -o yaml | grep NetworkTypeMigrationCompleted > /dev/null && \
+       for NODE in \$(oc get nodes -o custom-columns=NAME:.metadata.name --no-headers); do oc get node \$NODE -o yaml | grep "k8s.ovn.org/node-transit-switch-port-ifaddr:" | grep "100.85";  done > /dev/null && \
+       for NODE in \$(oc get nodes -o custom-columns=NAME:.metadata.name --no-headers); do oc get node \$NODE -o yaml | grep "k8s.ovn.org/node-gateway-router-lrp-ifaddr:" | grep "100.65";  done > /dev/null && \
+       oc get network.config/cluster -o jsonpath='{.status.networkType}' | grep OVNKubernetes > /dev/null;
+    do
+       echo "Live migration is still in progress"
+       sleep 300
+    done
+EOT
+    echo "The Migration is completed"
+    # Check all ClusterOperators back to normal after live migration
+    timeout 600s bash <<EOT
```

Comment:
we have similar function:
```
        if check_clusteroperators; then
            echo "Passed #${continous_successful_check}"
            (( continous_successful_check += 1 ))
        else
            echo "cluster operators are not ready yet, wait and retry..."
            continous_successful_check=0
        fi
```

Reply:
thanks, I reused `headlth_check` steps after migration.